src/web/client/common/remoteServiceProvider.ts

```diff
@@ -0,0 +1,147 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+'use strict';
+import * as vscode from 'vscode';
+import { getHeader } from './authenticationProvider';
+import { DEFAULT_LANGUAGE_CODE, pathparam_schemaMap, SINGLE_ENTITY_SAVE_URL_KEY, WEBPAGES_FILENAME } from './constants';
+import { CONTENTPAGES, PORTALSFOLDERNAME, PORTALSURISCHEME, SINGLE_ENTITY_FETCH_URL_KEY, WEBPAGES, WEBTEMPLATES } from './constants';
+import { PortalsFS } from './fileSystemProvider';
+import { entitiesSchemaMap, dataSourcePropertiesMap } from './localStore';
+
+export async function fetchData(accessToken: string, entity: string, entityId: string, queryParamsMap: any, entitiesSchemaMap: any, languageIdCodeMap: any, portalFs: PortalsFS) {
```

Comment:
at the moment this function has a hardcoded implementation. The implementation should be generic to handle single/multi file editing.